### PR TITLE
[Build] Handle license check and vetting request in one workflow

### PR DESCRIPTION
**Pull request by HannesWell**
This simplifies maintenance since the license handling is not distributed over multiple workflows.

### CODE REVIEW

**Review by laeubi — CHANGES_REQUESTED** [review]
@HannesWell actually this was intended, because the *check* itself requires less permissions, I assume this does not change that often anyways to become an issue.

**Comment by laeubi** [comment]
This seem to complicate the handling a lot just to save a duplicate line that almost never changes... 

Also I actually like it to be shown as part of the "Continuous Integration" workflow instead of a separate one.

File: .github/workflows/licensecheck.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
-name: Request license-review
+# This workflow will check for Maven projects if the licenses of all (transitive) dependencies are vetted.
+
+name: License vetting status check
```
Comment (laeubi) [inline .github/workflows/licensecheck.yml:3]: This is now misleading as it also handles the previous `Request license-review`

**Comment by HannesWell** [comment]
> This seem to complicate the handling a lot just to save a duplicate line that almost never changes...
>
Using one [ternary operator](
https://docs.github.com/en/actions/learn-github-actions/expressions#operators) does not seem to be a lot more complicated to me (tbh the GH workflow syntax is unknown also to me, but that's their decision).

On the other hand having the license handling distributed over multiple files also makes it more complicated to understand, even if does not change that often. Which weights more is a matter of preference.

> Also I actually like it to be shown as part of the "Continuous Integration" workflow instead of a separate one.

I don't think this is a big issue. The license check is a 'meta' check any ways not directly related to building a change.
And in the end in the vast majority of cases I just have a look at the check summary at the bottom of a PR and there all checks are listed in just one flat list. How it is present in the `Checks` tab of the PR or even in the `Actions` section of this repo is at least by myself not often looked at.

I understand the technical concerns about too elevated permission, but not those about the presentation.
Furthermore everyone else that I know of who is using this workflow is using it from a single file.

> 
> Actually it would even be better if they where two different workflow files for the different purposes...


Feel free to provide PRs to https://github.com/eclipse-dash/dash-licenses, but it's not that easy to achieve that without a major rework because many is shared in both cases. And in the end I don't think it is worth the effort.

File: .github/workflows/licensecheck.yml (hunk 1 above)
Comment (HannesWell) [reply .github/workflows/licensecheck.yml]: Then lets just name it `License status`.

**Comment by laeubi** [comment]
> I don't think this is a big issue. The license check is a 'meta' check any ways not directly related to building a change.

For me this is an important part, having the same workflow referenced in two places is something I way less often need to think about than looking at the actual checks.

So I still don't see any benefits here beside that now the check runs with elevated peremissions (PR write) and it feels confusing that the workflow performs two different tasks (one for checking license and one for requesting a review).

It would actually be better if the check itself would be split into two dedicated ones, one for checking the license and one for requesting a review, then there won't be any "duplication".

File: .github/workflows/licensecheck.yml (hunk 1 above)
Comment (laeubi) [reply .github/workflows/licensecheck.yml]: This is still wrong, requesting a review is not a status, also I suspect that now the check will maybe become green if requesting succeeds... I still see no benefit in mixing these two concerns into one workflow run that makes it more complex to manage and understand and test.

e.g. previously if the check failed one could rerun it, what will happen now is unclear, will it repeat the check of licenses or will it repeat the request for license and all that kind of uncertainties.

**Comment by laeubi** [comment]
> On the other hand having the license handling distributed over multiple files also makes it more complicated to understand, even if does not change that often.

We have two cases:

1. check license
2. request licensee review

so these are not "multiple places" it is exactly one place for one purpose... on the other hand now I need to understand that the shared workflow has two cases (check/request) and the workflow referenced also has different cases... so for me this still is not worth the hassle just because we have change the name once what most probably will never happen again.

**Comment by HannesWell** [comment]
> > I don't think this is a big issue. The license check is a 'meta' check any ways not directly related to building a change.
> 
> For me this is an important part, having the same workflow referenced in two places is something I way less often need to think about than looking at the actual checks.

Looks like this is the root of our disagreement. For me its the other way round.

> So I still don't see any benefits here beside that now the check runs with elevated peremissions (PR write) and it feels confusing that the workflow performs two different tasks (one for checking license and one for requesting a review).

What the workflow does or in how many files it is distributed in the end is probably irrelevant for most contributors since they only care if it works.
But the elevated permissions are indeed a drawback, therefore I'm closing this.

**Pull request closed without merging**